pkg/resolution/resolver/bundle/resolver_test.go

```diff
@@ -169,6 +191,335 @@ func TestResolveDisabled(t *testing.T) {
+			name: "single-task-using-configmap",
+			objs: []runtime.Object{
+				exampleTask,
+			},
+			mapper: test.DefaultObjectAnnotationMapper,
+			args: &args{
+				name: "example-task",
+				// use default kind value in configmap
+			},
+			expectedStatus: createResolutionRequestStatus(taskAsYAML),
+		}, {
+			name: "single-pipeline",
+			objs: []runtime.Object{
+				examplePipeline,
+			},
+			mapper: test.DefaultObjectAnnotationMapper,
+			args: &args{
+				name: "example-pipeline",
+				kind: "pipeline",
+			},
+			expectedStatus: createResolutionRequestStatus(pipelineAsYAML),
+		}, {
+			name: "multiple-resources",
+			objs: []runtime.Object{
+				exampleTask,
+				examplePipeline,
+			},
+			mapper: test.DefaultObjectAnnotationMapper,
+			args: &args{
+				name: "example-pipeline",
+				kind: "pipeline",
+			},
+			expectedStatus: createResolutionRequestStatus(pipelineAsYAML),
+		},
+		{
+			name:   "too-many-objs",
+			objs:   tooManyObjs,
+			mapper: test.DefaultObjectAnnotationMapper,
+			args: &args{
+				name: "2-task",
+				kind: "task",
+			},
+			expectedStatus:     createFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("contained more than the maximum %d allow objects", MaximumBundleObjects),
+		}, {
+			name:   "single-task-no-version",
+			objs:   []runtime.Object{&pipelinev1beta1.Task{TypeMeta: metav1.TypeMeta{Kind: "task"}, ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			mapper: asIsMapper,
+			args: &args{
+				name: "foo",
+				kind: "task",
+			},
+			expectedStatus:     createFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("the layer 0 does not contain a %s annotation", BundleAnnotationAPIVersion),
+		}, {
+			name:   "single-task-no-kind",
+			objs:   []runtime.Object{&pipelinev1beta1.Task{TypeMeta: metav1.TypeMeta{APIVersion: "tekton.dev/v1beta1"}, ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			mapper: asIsMapper,
+			args: &args{
+				name: "foo",
+				kind: "task",
+			},
+			expectedStatus:     createFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("the layer 0 does not contain a %s annotation", BundleAnnotationKind),
+		}, {
+			name:   "single-task-kind-incorrect-form",
+			objs:   []runtime.Object{&pipelinev1beta1.Task{TypeMeta: metav1.TypeMeta{APIVersion: "tekton.dev/v1beta1", Kind: "Task"}, ObjectMeta: metav1.ObjectMeta{Name: "foo"}}},
+			mapper: asIsMapper,
+			args: &args{
+				name: "foo",
+				kind: "task",
+			},
+			expectedStatus:     createFailureStatus(),
+			expectedErrMessage: fmt.Sprintf("the layer 0 the annotation %s must be lowercased and singular, found %s", BundleAnnotationKind, "Task"),
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a new image with the objects.
+			ref, err := test.CreateImageWithAnnotations(fmt.Sprintf("%s/testbundleresolver/%s", u.Host, tc.name), tc.mapper, tc.objs...)
+			if err != nil {
+				t.Fatalf("could not push image: %#v", err)
+			}
+			// set bundle param to be the image ref
+			tc.args.bundle = ref
```

Comment:
> It doesn't look like this is ever set in args. Let's just set it here.

Sounds good. I pre-populated the oci registry with images, and the ref is used for the args.bundle in the test table. PTAL.

> same with service account - it doesn't look like it's ever used

Yes, service account is not used in the test, but the param validation expects this to be set somewhere either from param or configmap, otherwise the param validation will fail. That's why we added a placeholder service account field in the configmap. PTAL and let me know wdyt. Thanks!